src/H5mpi.c

```diff
@@ -782,4 +782,58 @@ H5_mpio_gatherv_alloc_simple(void *send_buf, int send_count, MPI_Datatype send_t
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5_mpio_gatherv_alloc_simple() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5_mpio_get_file_sync_required
+ *
+ * Purpose:     Retrieve the MPI hint indicating whether the data written
+ *              by the MPI ROMIO driver is immediately visible to all MPI
+ *              ranks.
+ *
+ * Notes:       This routine is designed for supporting UnifyFS that needs
+ *              MPI_File_sync in order to make the written data available
+ *              to all ranks.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ * Programmer:  Houjun Tang,  April 7, 2022
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5_mpio_get_file_sync_required(MPI_File fh, hbool_t *file_sync_required)
+{
+    MPI_Info info_used;
+    int      flag;
+    char     value[MPI_MAX_INFO_VAL];
+    herr_t   ret_value = SUCCEED;
+
+    FUNC_ENTER_NOAPI(FAIL)
+
+    HDassert(file_sync_required);
+
+    *file_sync_required = FALSE;
+
+    if (MPI_SUCCESS != MPI_File_get_info(fh, &info_used))
+        HGOTO_ERROR(H5E_LIB, H5E_CANTGET, FAIL, "can't get MPI info")
+
+    if (MPI_SUCCESS !=
+        MPI_Info_get(info_used, "romio_visibility_immediate", MPI_MAX_INFO_VAL - 1, value, &flag))
```

Comment:
I think this flag leads now to the effect that NFS writes are also enabling the file sync, which crashes independent I/O:
https://github.com/openPMD/openPMD-api/issues/1423